PYME/recipes/tracking.py

```diff
@@ -125,12 +128,14 @@ def TrackWithPipeline(self, pipeline):
         into the pipeline (used when called from PYME.DSView.modules.particleTracking)"""
         clumpInfo, clumps = self.Track(pipeline)
 
-        pipeline.addColumn('clumpIndex', clumpInfo['clumpIndex'])
-        pipeline.addColumn('clumpSize', clumpInfo['clumpSize'])
-        pipeline.addColumn('trackVelocity', clumpInfo['trackVelocity'])
+        # pipeline.addColumn('clumpIndex', clumpInfo['clumpIndex'])
+        # pipeline.addColumn('clumpSize', clumpInfo['clumpSize'])
+        # pipeline.addColumn('trackVelocity', clumpInfo['trackVelocity'])
         
         #self.clumps = clumps        
-        pipeline.clumps = clumps
+        # pipeline.clumps = clumps
+
+        pipeline.addDataSource('clumps', clumpInfo)
```

Comment:
This won't work. The legacy mode should either remain as is, or (potentially): 

- create a new mapping filter from the `selectedDataSource`
- add the required columns
- select the new data mapping as the `selectedDataSource`

This might also not work, as this is not the full VisGUI pipeline. My preference would be to leave as is until we fix `DSView.modules.particleTracking`.

Reply:
It does work. I can't leave it because a postdoc in the Toomre lab wants to use this on Wednesday.

Reply:
I did the minimal changes needed to get the exact workflow you outlined in that PDF working again. I've run through the whole pipeline several times with the code in this PR, so I'm confident it does work.

Reply:
On second thought, I think I equated the pipeline working with this specific line working. It does add the dataSource, but we never actually use `pipeline.clumps` downstream in `particleTracking`, only the `clumps` `ClumpManager` that gets returned. So it's possible this specific line is not doing the right thing. It does not throw an error, though, whereas the `pipeline` injection did. Could just comment it out entirely.

Reply:
Confirmed that commenting it out entirely does still work.

Reply:
ahh, I was forgetting that the tracking module handles it's own overlays. 

I think it still breaks the (undocumented) option of opening a visgui pane in dsviewer and viewing the tracks there.

Reply:
I think this should be OK still, because I rewrote the tracks layer to automatically figure out if we're passing it a `ClumpManager` or a `tabular` tracks source. Switching up the data source type on PYMEVisualize shouldn't confuse it.

Reply:
But that might be broken anyway with recent VisGUI tracking changes.

Reply:
I could fix it if it is. I could be wrong about this, but I think the possibility that something else is already broken shouldn't preclude us from merging something that fixes a different part of the program that we know is broken. What needs to be changed here to get this merged so a postdoc here can use it?

Reply:
The might already be broken bit was me saying that it probably didn't matter. 

When I first read the code I thought that the track display within dsviewer was going to be broken (it used to be part of a very bloated arrayviewpanel). I'd forgotten I'd already factored it out into something saner. I'll try to get this in asap.

Reply:
Ah, sorry, I misunderstood. 

Thanks, David!

Reply:
Long-term, I think it would be nice to have `PYMEImage` use layers like `VisGUI`. We could update the annotation tools to live in an `Annotation` layer that sits on top of the image and tracks layers.

Reply:
You could actually pass the layer between `PYMEImage` and `PYMEVisualize`.

Reply:
Long term that is 100% the plan, and extends a bit further to making `visgui` the viewer for everything (using the ImageRenderLayer for image data). As it stands, `arrayviewpanel` does have lightweight layers of a sort which it loops through to draw some (but not all) of the overlays. The `particleTracking` module is using these at present, but these layers are completely separate from the VisGUI layers (use wx.DC drawing commands rather than OpenGL, and not as well organised as in VisGUI).

Reply:
I think it's even in the roadmap :p